src/tic_tac_toe/board.clj

```diff
@@ -1,11 +1,114 @@
 
 (defn place-mark
-  [position mark]
-  (assoc initial-board position mark))
+  [board position mark]
+  (assoc board position mark))
+
+(defn eq-mark-one-or-two
+  [mark]
+  (or (= mark player-one-mark) (= mark player-two-mark)))
+
+(defn is-full?
+  [board]
+  (every? (fn [mark] (eq-mark-one-or-two mark)) board))
+
+(defn top-row-winner?
+  [board]
+  (let [top-row [(get board 0) (get board 1) (get board 2)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn mid-row-winner?
+  [board]
+  (let [top-row [(get board 3) (get board 4) (get board 5)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn bot-row-winner?
+  [board]
+  (let [top-row [(get board 6) (get board 7) (get board 8)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn row-winner?
+  [board]
+  (cond 
+    (top-row-winner? board) (top-row-winner? board)
+    (mid-row-winner? board) (mid-row-winner? board)
+    :else (bot-row-winner? board)))
+
+(defn lft-col-winner?
+  [board]
+  (let [top-row [(get board 0) (get board 3) (get board 6)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn mid-col-winner?
+  [board]
+  (let [top-row [(get board 1) (get board 4) (get board 7)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn rt-col-winner?
+  [board]
+  (let [top-row [(get board 2) (get board 5) (get board 8)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn col-winner?
+  [board]
+  (cond 
+    (lft-col-winner? board) (lft-col-winner? board)
+    (mid-col-winner? board) (mid-col-winner? board)
+    :else (rt-col-winner? board)))
+
+(defn toplft-diag-winner?
+  [board]
+  (let [top-row [(get board 0) (get board 4) (get board 8)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
+(defn botlft-diag-winner?
+  [board]
+  (let [top-row [(get board 6) (get board 4) (get board 2)]]
+    (cond
+      (every? (fn [mark] (= player-one-mark mark)) top-row) player-one-mark
+      (every? (fn [mark] (= player-two-mark mark)) top-row) player-two-mark
+      :else false)))
+
```

Comment:
So, maybe you could create all the rows (horizontal, vertical, and diagonals) in a separated method, and then you could use the extracted method `winner?` I've suggested further above (which you actually already have below...) just once and iterate through the list of all rows to check if any of them is a winner? Does this make sense?!